src/client.js

```diff
@@ -75,7 +75,7 @@ let getRegistration = async function() {
 
 class PushClient {
   constructor({endpointUrl=null, userId=null, workerUrl=WORKER_URL,
-      scope=SCOPE}) {
+      scope=SCOPE} = {}) {
```

Comment:
Yes, sorry, hadn't noticed that this was going into an unrelated PR :)

Check out [this sample](https://babeljs.io/repl/#?experimental=false&evaluate=true&loose=false&spec=false&code=let%20a%20%3D%20function(%7Bp%3D1%7D)%20%7B%20console.log(p)%3B%20%7D%0A%0Alet%20b%20%3D%20function(%7Bp%3D1%7D%20%3D%20%7B%7D)%20%7B%20console.log(p)%3B%20%7D%0A%0A%0Ab()%3B%0Aa()%3B)

You can see that without defaulting the whole parameter to an Object, `new PushClient()` would cause all of the values to be undefined. With it you get the defaults applied.